src/caterpillar.jl

```diff
@@ -16,27 +16,34 @@ struct RanefInfo{T<:AbstractFloat}
 end
 
 """
-    ranefinfo(m::LinearMixedModel)
+    ranefinfo(m::MixedModel)
 
 Return a `NamedTuple{fnames(m), NTuple(k, RanefInfo)}` from model `m`
 """
-function ranefinfo(m::LinearMixedModel{T}) where {T}
+function ranefinfo(m::MixedModel{T}) where {T}
     fn = fnames(m)
     val = sizehint!(RanefInfo[], length(fn))
-    for (re, eff, cv) in zip(m.reterms, ranef(m), condVar(m))
-        push!(
-            val,
-            RanefInfo(
-                re.cnames,
-                re.levels,
-                Matrix(adjoint(eff)),
-                Matrix(adjoint(dropdims(sqrt.(mapslices(diag, cv, dims=1:2)), dims=2))),
-            )
-        )
+    for grp in fn
+        push!(val, ranefinfo(m, grp))
     end
     NamedTuple{fn}((val...,))
 end
 
+"""
+    ranefinfo(m::MixedModel, gf::Symbol)
+
+Return a `RanefInfo` corresponding to the grouping variable `gf` model `m`.
+"""
+function ranefinfo(m::MixedModel, gf::Symbol)
+    idx = findfirst(==(gf), fnames(m))
+    idx !== nothing || throw(ArgumentError("$gf is not the name of a grouping variable in the model"))
+
+    re, eff, cv = m.reterms[idx], ranef(m)[idx], condVar(m)[idx]
```

Comment:
If we are going to do it this way we should provide methods for `ranef` and `condVar` that take a grouping factor symbol as a second argument.  As it stands `ranefinfo(m)` will evaluate `ranef` and `condVar` for all the grouping factors multiple times.  I defined a `condVar(m, fn)` method in the `db/ranefL` branch of `MixedModels.jl`.  Did we discuss whether that would need a patch release or a minor release?

Reply:
Good catch!

Adds a feature in post 1.0: minor release.